### PR TITLE
Bugfix #296 added delete component feature

### LINKED ISSUE #296: API Call: Delete Component not working anymore
**Describe the bug**
* "Cleanup" method is being used by ProvApp REST Api to delete Projects. 
* Because Jenkins calls are ASYC JenkinsPipelineAdapter cant do cleanup as before 
( Cleanup job can be executed before actual project creation happens) -> Cleanup procedure was disabled in JenkinsPipelineAdapter 

**To Reproduce**
call REST API to delete Project

**Expected behavior**
Openshift compontens based on quickstarters are being deleted

***TODO:***
- [x]  create Jenkinsjob in ods-core to delete components from openshift (-> @kiwo )
- [x]  build "delete-component" Function in JenkinsPipelineAdapter (create a new function desigend to delete components. (-> @stefanlack )
- [x] Activate and fix test E2EProjectAPIControllerTest.testQuickstarterProvisionOnNewOpenProjectInclDeleteSingleComponent  (based on #295)/ function that is called to delete a single component is missing (-> @stefanlack )

***Additional Context***

Code where _delete compontent function_ was removed: https://github.com/opendevstack/ods-provisioning-app/commit/86e56adb5d85ac07c01fd312e05fdc06577e0d13#r36243487

### PR DESCRIPTION
**Pull request by stefanlack**
fixes #296

### CODE REVIEW

**Comment by stefanlack** [comment]
Status:
- implementation and local test ready
- Was yet not tested against running OpenDevStack in openshift

File: ocp-config/prov-app/cm.yml [hunk 1]
```diff
@@ -138,12 +138,15 @@ objects:
...
-      jenkinspipeline.quickstarter.create-projects.type=project
-
-      # Backend quickstarters
+      #Adminjob configuration: create projects, delete-projects and delete-components
+      jenkinspipeline.adminjobs.create-projects.desc=Create Project
+      jenkinspipeline.adminjobs.create-projects.url=opendevstack/ods-core.git#${ODS_GIT_REF}/create-projects/Jenkinsfile
+      jenkinspipeline.adminjobs.delete-projects.desc=Delete openshift project
+      jenkinspipeline.adminjobs.delete-projects.url=opendevstack/ods-core.git#${ODS_GIT_REF}/delete-projects/Jenkinsfile
+      jenkinspipeline.adminjobs.delete-components.desc=Delete openshift components
+      jenkinspipeline.adminjobs.delete-components.url=opendevstack/ods-core.git#${ODS_GIT_REF}/delete-components/Jenkinsfile
```
Comment (michaelsauter) [inline ocp-config/prov-app/cm.yml]: Like the new name "adminjobs" instead of "quickstarters" 👍

File: ocp-config/prov-app/cm.yml [hunk 2]
```diff
@@ -157,15 +160,15 @@ objects:
...
+      # Data Science quickstarter
       jenkinspipeline.quickstarter.dsJupyterNotebook.desc=Data Science - Jupyter Notebook
       jenkinspipeline.quickstarter.dsJupyterNotebook.url=opendevstack/ods-quickstarters.git#${ODS_GIT_REF}/ds-jupyter-notebook/Jenkinsfile
       jenkinspipeline.quickstarter.dsRshiny.desc=Data Science - R-Shiny
       jenkinspipeline.quickstarter.dsRshiny.url=opendevstack/ods-quickstarters.git#${ODS_GIT_REF}/ds-rshiny/Jenkinsfile
       jenkinspipeline.quickstarter.dsMlService.desc=Data Science - Machine Learning
       jenkinspipeline.quickstarter.dsMlService.url=opendevstack/ods-quickstarters.git#${ODS_GIT_REF}/ds-ml-service/Jenkinsfile
 
-      # Frontend quickstarters
+      # Frontend Quickstarter
```
Comment (michaelsauter) [inline ocp-config/prov-app/cm.yml]: 😞 looks my typo fixes get reverted here?

File: ocp-config/prov-app/cm.yml (hunk 1 above)
Comment (stefanlack) [reply ocp-config/prov-app/cm.yml]: Thanks.

File: ocp-config/prov-app/cm.yml (hunk 2 above)
Comment (stefanlack) [reply ocp-config/prov-app/cm.yml]: Done.

**Pull request merged**